packages/liveblocks-core/src/lib/autoRetry.ts

```diff
@@ -1,6 +1,20 @@
 import * as console from "./fancy-console";
+import type { JsonObject } from "./Json";
 import { wait } from "./utils";
 
+export class HttpError extends Error {
+  constructor(
+    public message: string,
+    public status: number,
+    public details?: JsonObject
+  ) {
+    super(message);
+  }
+}
```

Comment:
Should this live elsewhere maybe? Since `autoRetry` isn't `fetch`-specific. Same for the default `shouldStopRetrying` value, in addition to not retrying 4xx maybe it could also not retry `AbortError` and `TimeoutError` errors to support `AbortSignal`-powered APIs out-of-the-box.